server/controllers/post.js

```diff
@@ -43,3 +46,115 @@ exports.creatForumPost = async (req, res, next) => {
         next(error);
     }
 };
+
+
+exports.deletePost = async (req, res, next) => {
+    const { forumId, postId } = req.params;
+
+    try {
+        const forum = await Forum.findById(forumId);
+        if (!forum) throw new NotFound('No forum found');
+
+        const post = await Post.findById(postId);
+        if (!post) throw new NotFound('No post found');
+
+        const forumPost = forum.posts.find(fp => fp.toString() === postId);
+        if (!forumPost) throw new NotFound('No post in forum found');
+
+        // check if useres are deleteing their own posts
+        if (post.user.toString() !== req.body.userId)
+            throw new Unauthorized('Cannot delete post');
+
+        forum.posts = forum.posts.filter(fp => fp.toString() !== postId);
+
+        const forumRes = await forum.save();
+        if (!forumRes) throw new GeneralError('Error saving forum');
+
+        const postRes = await post.remove();
```

Comment:
I think we had discussed that when "deleting" the post we should only prevent it from being viewed, instead of actually removing the post from the DB.

You can accomplish this by adding an 'isDeleted' flag to the Post model, and filtering those results out when fetching posts

Reply:
Added "isDeleted" flag to post and comment models. So, now fetch will only return those that have "isDeleted" flag set to false.